### ISSUE TITLE
Adding hc-navbar-mobile-menu requires app switcher to be included

### ISSUE THREAD

**Issue by jonest** [HealthCatalyst/Fabric.Cashmere#280]
I'm attempting to switch over Ops Console to use Cashmere piece by piece and when I switch the navbar, I can leave out the app switcher. When I then add in the mobile menu, it requires that I include the app switcher stuff, even if I set `appSwitcher="false"`.

```
AppComponent.html:11 ERROR Error: StaticInjectorError[IAppSwitcherService]: 
  StaticInjectorError[IAppSwitcherService]: 
    NullInjectorError: No provider for IAppSwitcherService!
    at _NullInjector.webpackJsonp.../../../core/esm5/core.js._NullInjector.get (core.js:923)
    at resolveToken (core.js:1211)
    at tryResolveToken (core.js:1153)
    at StaticInjector.webpackJsonp.../../../core/esm5/core.js.StaticInjector.get (core.js:1024)
    at resolveToken (core.js:1211)
    at tryResolveToken (core.js:1153)
    at StaticInjector.webpackJsonp.../../../core/esm5/core.js.StaticInjector.get (core.js:1024)
    at resolveNgModuleDep (core.js:10586)
    at NgModuleRef_.webpackJsonp.../../../core/esm5/core.js.NgModuleRef_.get (core.js:11807)
    at resolveDep (core.js:12303)
```

**Comment by jonest** [comment]
I believe this was addressed recently when @benjanderson pushed up a new package to npm. Is this correct?

**Comment by benjanderson** [comment]
I can confirm this, and it is not fixed in 0.0.9
https://stackblitz.com/edit/cashmere-demo-wout-app-switcher?file=app%2Fnavbar%2Fnavbar.component.html

I think this is a bug :bug:

**Comment by jonest** [comment]
Ah, you're correct. I was mixing this issue up with the one where some components weren't included until that recent update :)

**Comment by jonest** [comment]
From issue #340 :

https://stackblitz.com/edit/cashmere-demo-a69f58

On this particular page ([navbar.component.html](https://stackblitz.com/edit/cashmere-demo-a69f58?file=app%2Fnavbar%2Fnavbar.component.html,`)) if you uncomment the `hc-navbar-mobile-menu` it will no longer load the page correctly

**Comment by jtrujill** [comment]
It seems to make more sense to extract appswitcher out into it's own directive. Then whatever element has the appswitcher applied to it get triggered when a user has clicked on that element.

I haven't looked at the source code for iappswitcherservice but does it make sense for it to be an interface? Could it not take in a config object that tells it how to find the different services?

**Comment by benjanderson** [comment]
app switcher is its own component and I believe the dependency was added when we added the mobile capabilities into the navbar and had a "View all my applications" buttons. I am working on a solution...

![image](https://user-images.githubusercontent.com/1934453/41805298-8015ca3a-7664-11e8-91c2-45d6320a2912.png)

the setup is the following, making it so navbar doesn't really need to know about the app switcher component
```
<hc-navbar ...>
<hc-app-switcher></hc-app-switcher>
...
```

As for interface vs config, the dependency injection is setup by default in the AppSwitcherModule and you only have to override the provided interface if you so choose. So the following is the only thing the consumer has to pass in
```
@NgModule({
    imports: [
        AppSwitcherModule.forRoot({discoveryServiceUri: 'http://SERVER/DiscoveryService'})
    ],
...
```

I like the idea of not crashing if its not provided and am working on a solution for that. Ss we talked with Ty earlier, he didnt have the discovery service setup so he wanted to use the MockAppSwitcherService which would require something like 
```
import {NavbarModule, PopoverModule, AppSwitcherModule, IconModule, MockAppSwitcherService, ListModule} from '@healthcatalyst/cashmere';

@NgModule({
    imports: [NavbarModule, PopoverModule, AppSwitcherModule, IconModule],
    providers: [
        {
            provide: 'IAppSwitcherService',
            useClass: MockAppSwitcherService
        }
    ]
})
export class AppModule {}
```

This was all in the mark down that was deleted, but I am currently working on integrating it with the components so dgeni documents it
